src/Microsoft.Health.Dicom.Core/Features/Indexing/IDicomDatasetReindexValidator.cs

```diff
@@ -0,0 +1,25 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Dicom;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.Indexing
+{
+    /// <summary>
+    /// Validator that validates DicomDataset for reindexing.
+    /// </summary>
+    public interface IDicomDatasetReindexValidator
```

Comment:
Why do we need another class to validate this? Is'nt this same as STOW validation which we already do with ExtendedQueryTag.

Reply:
there are some difference between them. 
1. STOW validates all 3 Uids (exists and different), and some required tag (patientId, SopClassUid), while reindex don't need to
2. STOW support full validation, while reindex don't need to
3. STOW validate all query tags while reindex is on subset (this could be solved by refactoring)
4. STOW throws exception on failure, while reindex log the error (this could also be solved by refactoring)

Reply:
I was trying merge these 2 validators together, but find the code is fragile, so use a different way to simplify.
* Encapsulate STOW validation into 3 classes -- `DatasetCoreTagsValidation`, `DatasetFullValidation` and `DatasetQueryTagsValidation`
* Reindex validation also calls `DatasetQueryTagsValidation`
* `DatasetQueryTagsValidation` accept custom `Function` to decided whether throws exceptions or not ( STOW throws exception while Reindex not throw but write into database)
* Also rename:  `DicomDatasetValidator` => `StoreDatasetValidator` , `DicomDatasetReindexValidator`=>`ReindexDatasetValidator` , which I feel more clear.

Please let me know if you have any concerns.

Reply:
Discussed with Will, the 3 validation classes (DatasetCoreTagsValidation, DatasetFullValidation and DatasetQueryTagsValidation) is not appropriate based on their usage (new class , call function and discard), method would be  better.
Updated code based on the discussion. Get rid of the 3 classes.